packages/compass-crud/src/stores/crud-store.ts

```diff
@@ -1584,6 +1600,72 @@ class CrudStoreImpl
   openCreateSearchIndexModal() {
     this.localAppRegistry.emit('open-create-search-index-modal');
   }
+
+  openBulkDeleteDialog() {
+    const PREVIEW_DOCS = 5;
+
+    this.setState({
+      bulkDelete: {
+        previews: this.state.docs?.slice(0, PREVIEW_DOCS) || [],
+        status: 'open',
+        affected: this.state.count || 0,
+      },
+    });
+  }
+
+  bulkDeleteInProgress() {
+    this.setState({
+      bulkDelete: {
+        ...this.state.bulkDelete,
+        status: 'in-progress',
+      },
+    }); // TODO: COMPASS-7328
+  }
+
+  bulkDeleteFailed(ex: Error) {
+    return ex; // TODO: COMPASS-7328
+  }
+
+  bulkDeleteSuccess() {
+    this.closeBulkDeleteDialog();
+    // TODO: COMPASS-7328
+  }
+
+  closeBulkDeleteDialog() {
+    this.setState({
+      bulkDelete: {
+        ...this.state.bulkDelete,
+        status: 'closed',
+      },
+    });
+  }
+
+  async runBulkDelete() {
+    const { affected } = this.state.bulkDelete;
+    this.closeBulkDeleteDialog();
```

Comment:
If we close this and user clicks cancel while confirming, should we show it back? This (hiding delete modal and show confirmation modal) creates a flicker. Maybe we should keep showing the delete modal as well?

Reply:
Discussed on slack, but writing the answer here for reference:

This is as discussed with design, we are not going to use double modals for this flow. We might want to revisit later.